identity/src/main/java/com/android/identity/util/SimpleApplicationData.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.android.identity.internal.Util;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.CborDecoder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.builder.MapBuilder;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.UnicodeString;
+
+/**
+ * An implementation of {@link ApplicationData} using a LinkedHashMap to back the key-value pairs.
+ */
+public class SimpleApplicationData implements ApplicationData {
+    private Map<String, byte[]> mApplicationData = new LinkedHashMap<>();
+    private Callable<Boolean> mSaveFunc = () -> { return false; };
+
+    /**
+     * Creates a new SimpleApplicationData.
+     */
+    public SimpleApplicationData() {
+
+    }
+
+    @Override
+    public Set<String> getKeySet() {
+        return new LinkedHashSet<>(mApplicationData.keySet());
+    }
+
+    @Override
+    public void setSaveFunc(Callable<Boolean> saveFunc) {
+        mSaveFunc = saveFunc;
+    }
+
+    @Override
+    public void setApplicationData(@NonNull String key, @Nullable byte[] value) {
+        if (value == null) {
+            mApplicationData.remove(key);
+        } else {
+            mApplicationData.put(key, value);
+        }
+
+        try {
+            mSaveFunc.call();
+        } catch (Exception e) {
+            Logger.d("SimpleApplicationData", "Unexpected exception occurred: " + e);
+        }
```

Comment:
This seems unnecessary and a side-effect of using `Callable`. If you just use a listener instead, this is unnecessary.